ql/src/test/org/apache/hadoop/hive/ql/io/thrift/twitter/TestThriftGeneralDeserializer.java

```diff
@@ -0,0 +1,221 @@
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.StructField;
+import org.apache.hadoop.hive.serde2.typeinfo.StructTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.thrift.TBase;
+import org.apache.thrift.TEnum;
+import org.apache.thrift.TFieldIdEnum;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.transport.TMemoryBuffer;
+
+import com.twitter.elephantbird.mapreduce.io.ThriftWritable;
+
+import junit.framework.TestCase;
+
+public class TestThriftGeneralDeserializer extends TestCase {
+
+  public void testThriftDeserializer() throws Throwable {
+    try {
+      // Create the thrift writable object
+      Tweet tweet = new Tweet(1, "newUser", "hello world")
+          .setLoc(new Location(1234, 5678))
+          .setAge((short) 26)
+          .setB((byte) 10)
+          .setIsDeleted(false)
+          .setTweetType(TweetType.REPLY)
+          .setFullId(1234567)
+          .setPic("abc".getBytes())
+          .setAttr(ImmutableMap.of("a", "a"));
+      TMemoryBuffer transport = new TMemoryBuffer(4096);
+      TBinaryProtocol protocol = new TBinaryProtocol(transport);
+      tweet.write(protocol);
+
+      // Create the SerDe
+      System.out.println("test: testThriftDeserializer");
+
+      final ThriftGeneralDeserializer serDe = new ThriftGeneralDeserializer();
+      final Configuration conf = new Configuration();
+      final Properties tbl = createProperties();
+      SerDeUtils.initializeSerDe(serDe, conf, tbl, null);
+
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      ThriftWritable writable = ThriftWritable.newInstance(Tweet.class);
+      writable.set(tweet);
+      writable.write(new DataOutputStream(out));
+
+      DataInput in = new DataInputStream(new ByteArrayInputStream(out.toByteArray()));
+      ThriftWritable newWritable = ThriftWritable.newInstance(ThriftGenericRow.class);
+      newWritable.readFields(in);
+
+      runTests(serDe, newWritable, tweet);
+
+    } catch (final Throwable e) {
+      e.printStackTrace();
+      throw e;
+    }
+  }
+
+  private void runTests(final ThriftGeneralDeserializer serDe, final ThriftWritable writable, Object tweet) throws SerDeException {
+
+    // Deserialize
+    final Object row = serDe.deserialize(writable);
+    assertEquals("deserialization gives the wrong object class", row.getClass(), ThriftGenericRow.class);
+    assertEquals("size incorrect after deserialization", serDe.getSerDeStats().getRawDataSize(), ((ThriftGenericRow) writable.get()).length());
+
+    // Lazy Field Deserialize
+    ThriftStructObjectInspector oi = (ThriftStructObjectInspector)serDe.getObjectInspector();
+    assertEquals("This test should pass", true, deserializeAndTestAllFields(row, oi, tweet, true));
+
+    Tweet tweet1 = new Tweet(1, "newUser", "hello world")
```

Comment:
I think you need to have another method that tests this `tweet1`, as you have done for `tweet`.